cms/migrations/0028_setup_course_program_index_pages.py

```diff
@@ -0,0 +1,102 @@
+"""
+Data migration to ensure the correct state for course/program index pages and
+correct depth for course/program detail pages
+"""
+from django.db import migrations
+from wagtail.core.models import Page
+
+COURSE_INDEX_PAGE_PROPERTIES = dict(title="Courses")
+PROGRAM_INDEX_PAGE_PROPERTIES = dict(title="Programs")
+
+
+def delete_wagtail_pages(specific_page_cls, filter_dict=None):
+    """
+    Completely deletes Wagtail CMS pages that match a filter. Wagtail overrides standard delete functionality,
+    making it difficult to actually delete Page objects and get information about what was deleted.
+    """
+    page_ids_to_delete = specific_page_cls.objects.values_list("id", flat=True)
+    if filter_dict:
+        page_ids_to_delete = page_ids_to_delete.filter(**filter_dict)
+    num_pages = len(page_ids_to_delete)
+    base_pages_qset = Page.objects.filter(id__in=page_ids_to_delete)
+    if not base_pages_qset.exists():
+        return 0, {}
+    base_pages_qset.delete()
+    return (
+        num_pages,
+        {specific_page_cls._meta.label: num_pages},  # pylint: disable=protected-access
+    )
+
+
+def get_top_level_wagtail_page():
+    """
+    The Wagtail CMS (at least in our usage) has one root page at depth 1, and one page at depth 2. All pages that we
+    create in Wagtail are added as children to the page at depth 2.
```

Comment:
Rather than going through pages at depth 1 or 2, I'm changing this to use the `Site` model to fish out the configured home page ID. That gets rid of all assumptions.